### ISSUE TITLE
Define StemID

### ISSUE THREAD

**Issue by theophoric** [stems/stem#1]

**Comment by jovincroninwilesmith** [comment]
#### Notes from Slack discussion
+ "Create a number sequence for stemID which would represent the artist, the label or institution associated with the asset, and the owner (MCN, partner, aggregator)"
+ "For example, everything coming from BPM the stem id would start with BPM, then if it was the protocol label it would be BPMPRO then if it was Don Diablo it would be BPMPRODD and the last 4 digits would represent the number of the asset in the system so BPMPRODD0001"
+ **Potential Risk:** High collision chance due to limited number of identifiers

**Comment by theophoric** [comment]
The only absolute requirement is that the identifier be unique within our system.  beyond that, it is desirable that the raw stemid encode some additional identifying information regarding the asset, such as the artist or label name.

**Comment by theophoric** [comment]
The format for the StemID should not change.  Information that is encoded in the stemId should be relevant and unchanged in the future.

**Comment by jovincroninwilesmith** [comment]
![screen shot 2015-04-22 at 11 09 36 pm](https://cloud.githubusercontent.com/assets/4871208/7289904/ac7a95ac-e944-11e4-9658-d72d85145df9.png)
An example structure seems like it could work and allow for a lot of options.

**Comment by theophoric** [comment]
questions i have:  
- will every asset have a label?
- will every asset have an artist?
- will the label for an asset ever change?
- will the artist for an asset ever change?

**Comment by theophoric** [comment]
What about encoding the audio fingerprint?

**Comment by theophoric** [comment]
We should also try to define what we mean by "asset" in this context. For what do we generate a StemId ?  Is it the actual recording?  Does it map directly to a youtube Asset ?  How will this change when we introduce more services and integrations ?

**Comment by milanastem** [comment]
no, not every asset will have a label. if that is missing, can we just use 0's?
yes, every asset will have an artist
no, the label will likely never change, however, the MCN may change as sometimes artists leave one MCN and join another.
no, the artist for the asset will never change.

**Comment by theophoric** [comment]
How will each artist be uniquely identified ?

**Comment by theophoric** [comment]
What are the objectives for having a human-readable stemid?  Would it suffice to just generate it from metadata in the reports?

**Comment by milanastem** [comment]
re: defining asset - please log into bpms@stem.is and read this: https://support.google.com/youtube/answer/6085468?hl=en&ref_topic=6084219

**Comment by jovincroninwilesmith** [comment]
StemID may not need to be displayed in our system, but could be created in human readable format when a report is generated.

**Comment by timluckow** [comment]
Jovin, that example structure is great and very on point. I think each StemID will need to compile several unique IDs... one for the artist + each collaborator artistID, one for the label/mcn, and one for the asset as described above. In addition to that, I believe we'll want to link platform IDs (SoundCloud/YouTube/etc)

Theo- artists connected to an asset won't change but eventually we might/will have derivative works (ContentID on YouTube) which is similar to the sub-contract structure, so we'd be keeping the original ownership structure and then sub-adding whoever created a new asset using a piece of the existing one. 

Labels/Publishers have actually been buying catalogues much more often and I think we should prepare for that scenario, essentially being able to switch label IDs for an entire batch of assets if their catalogue is acquired. This won't be necessary at the beginning but I can almost guarantee it will become a factor that can give us a leg up on the existing infrastructure at some point down the road. 

Having the ID readable by humans is probably not necessary, what The Orchard and some other distributors do is they give a catalogue # field so if for example BPM wants to have their own ID system for their CSVs, they can enter that in the field when uploading. Something that could be cool is if we let users set their own rules for cataloging within Stem and we attach that secondary catalogue # to the asset, keeping it separate from StemID but enabling them to process their CSV more easily.
